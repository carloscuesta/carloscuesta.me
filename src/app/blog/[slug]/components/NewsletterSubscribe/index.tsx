const NewsletterSubscribe = () => (
  <form
    className="my-8"
    action="https://buttondown.email/api/emails/embed-subscribe/carloscuesta"
    method="post"
    target="popupwindow"
  >
    <p className="pb-6 text-2xl font-semibold">Subscribe 👨‍💻</p>

    <div className="pb-6 text-lg">
      <label htmlFor="bd-email">
        Be the first to know when I write something new! If you don&#39;t like
        emails, you can follow me on{' '}
        <a
          className="underline underline-offset-2"
          href="https://twitter.com/intent/follow?screen_name=crloscuesta"
          rel="noopener noreferrer"
          target="_blank"
        >
          twitter
        </a>
        .
      </label>
    </div>

    <div>
      <div className="grid grid-flow-col grid-cols-[70%,30%] gap-3 sm:grid-cols-[312px,min-content]">
        <input
          id="bd-email"
          name="email"
          placeholder="you@email.com"
          required
          type="email"
          className="hover:text-neutral-1000 rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-2 py-1 text-neutral-700 opacity-100 outline-none ring-neutral-300  transition-all hover:border-opacity-50 hover:shadow-sm focus:ring-2 dark:border-neutral-700 dark:text-neutral-200 dark:ring-neutral-500 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2 "
        />

        <input type="hidden" value="1" name="embed" />

        <input
          type="submit"
          value="Subscribe"
          className="hover:text-neutral-1000 shadow-xs cursor-pointer rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-1 font-semibold text-neutral-700 outline-none ring-neutral-300 transition-all hover:border-opacity-50 hover:shadow-sm focus:ring-2 dark:border-neutral-700 dark:text-neutral-200 dark:ring-neutral-500 dark:hover:border-neutral-600 dark:hover:text-white"
        />
      </div>
    </div>
  </form>
)

export default NewsletterSubscribe
