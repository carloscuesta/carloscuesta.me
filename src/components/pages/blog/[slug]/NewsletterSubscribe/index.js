// @flow
import React from 'react'

import styles from './styles.module.css'

const NewsletterSubscribe = () => (
  <form
    className={styles.container}
    action='https://buttondown.email/api/emails/embed-subscribe/carloscuesta'
    method='post'
    target='popupwindow'
  >
    <b className={styles.title}>Subscribe to the newsletter 👨‍💻</b>

    <div className='row'>
      <div className='col-xs-12 col-sm-6'>
        <p className={styles.explanation}>
          Be the first to know when I write something new! If you don't like emails, you can
          follow me on {' '}
          <a href='https://twitter.com/intent/follow?screen_name=crloscuesta' rel='noopener noreferrer' target='_blank'>
            twitter
          </a>.
        </p>
      </div>

      <div className='col-xs-12 col-sm-6'>
        <p>
          <label htmlFor='bd-email'>Write your email:</label>
        </p>

        <div className={styles.form}>
          <input
            id='bd-email'
            className={styles.input}
            name='email'
            placeholder='Your email address'
            required
            type='email'
          />

          <input type='hidden' value='1' name='embed' />

          <input className={styles.submit} type='submit' value='Subscribe' />
        </div>
      </div>
    </div>
  </form>
)

export default NewsletterSubscribe
