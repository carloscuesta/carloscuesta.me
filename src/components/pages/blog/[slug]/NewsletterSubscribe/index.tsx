const NewsletterSubscribe = () => (
  <form
    className='my-8'
    action='https://buttondown.email/api/emails/embed-subscribe/carloscuesta'
    method='post'
    target='popupwindow'
  >
    <p className='text-2xl font-semibold pb-6'>Subscribe 👨‍💻</p>
    
    <div className='text-lg pb-6'>
      <label htmlFor='bd-email'>
        Be the first to know when I write something new! If you don&#39;t like emails, you can
        follow me on {' '}
        <a className='underline underline-offset-2' href='https://twitter.com/intent/follow?screen_name=crloscuesta' rel='noopener noreferrer' target='_blank'>
          twitter
        </a>.
      </label>
    </div>

    <div>
      <div className='grid grid-flow-col gap-3 grid-cols-[70%,30%] sm:grid-cols-[312px,min-content]'>
        <input
          id='bd-email'
          name='email'
          placeholder='you@email.com'
          required
          type='email'
          className='outline-none transition-all focus:ring-2 ring-neutral-300 dark:ring-neutral-500 sm:space-x-2 px-2 py-1 opacity-100 rounded-md text-neutral-700 hover:text-neutral-1000  bg-transparent border border-neutral-400 border-opacity-30 dark:border-neutral-700 dark:hover:border-neutral-600 dark:text-neutral-200 dark:hover:text-white hover:border-opacity-50 hover:shadow-sm '
        />

        <input type='hidden' value='1' name='embed' />

        <input type='submit' value='Subscribe' className='outline-none transition-all ring-neutral-300 dark:ring-neutral-500 focus:ring-2 font-semibold px-4 py-1 rounded-md text-neutral-700 hover:text-neutral-1000 shadow-xs bg-transparent border border-neutral-400 border-opacity-30 dark:border-neutral-700 dark:hover:border-neutral-600 dark:text-neutral-200 dark:hover:text-white hover:border-opacity-50 hover:shadow-sm cursor-pointer' />
      </div>
    </div>
  </form>
)

export default NewsletterSubscribe
