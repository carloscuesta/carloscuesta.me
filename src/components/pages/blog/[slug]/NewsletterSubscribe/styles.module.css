.container {
  border-radius: 4px;
  box-shadow: 0 1px 4px var(--shadowColor);
  margin: 1.5em 0;
  padding: 2em;
}

.title {
  font-size: 1.5em;
}

.explanation {
  margin-bottom: 0;
}

.form {
  display: flex;
}

.input {
  border-radius: 4px;
  border: 1px solid var(--text);
  flex: 1;
  font-family: inherit;
  font-size: 1rem;
  margin-right: .5em;
  padding: .5em;
}

.input:focus {
  outline-color: var(--secondary);
  border-color: var(--secondary);
}

.submit {
  -webkit-appearance: none;
  background-color: var(--secondary);
  border-radius: 4px;
  border: 0;
  color: var(--textOnSecondary);
  cursor: pointer;
  font-family: inherit;
  font-size: 1rem;
  padding: .5em;
  padding: .75em;
}

@media (prefers-color-scheme: dark) {
  .container {
    background-color: var(--backgroundSurface);
    box-shadow: none;
  }

  .input {
    background-color: var(--backgroundSurface);
    color: var(--text);
  }
}
