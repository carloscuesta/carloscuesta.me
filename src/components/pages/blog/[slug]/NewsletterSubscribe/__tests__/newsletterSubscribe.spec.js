import renderer from 'react-test-renderer'

import NewsletterSubscribe from '../index'

describe('NewsletterSubscribe', () => {
  it('should match NewsletterSubscribe component', () => {
    const wrapper = renderer.create(<NewsletterSubscribe />)

    expect(wrapper).toMatchSnapshot()
  })
})
