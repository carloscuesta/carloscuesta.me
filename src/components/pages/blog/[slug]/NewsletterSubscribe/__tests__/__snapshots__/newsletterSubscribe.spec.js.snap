// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsletterSubscribe should match NewsletterSubscribe component 1`] = `
<form
  action="https://buttondown.email/api/emails/embed-subscribe/carloscuesta"
  className="container"
  method="post"
  target="popupwindow"
>
  <b
    className="title"
  >
    Subscribe to the newsletter üë®‚Äçüíª
  </b>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-6"
    >
      <p
        className="explanation"
      >
        Be the first to know when I write something new! If you don't like emails, you can follow me on 
         
        <a
          href="https://twitter.com/intent/follow?screen_name=crloscuesta"
          rel="noopener noreferrer"
          target="_blank"
        >
          twitter
        </a>
        .
      </p>
    </div>
    <div
      className="col-xs-12 col-sm-6"
    >
      <p>
        <label
          htmlFor="bd-email"
        >
          Write your email:
        </label>
      </p>
      <div
        className="form"
      >
        <input
          className="input"
          id="bd-email"
          name="email"
          placeholder="Your email address"
          required={true}
          type="email"
        />
        <input
          name="embed"
          type="hidden"
          value="1"
        />
        <input
          className="submit"
          type="submit"
          value="Subscribe"
        />
      </div>
    </div>
  </div>
</form>
`;
